src/profiler.cpp

```diff
@@ -692,6 +692,36 @@ u64 Profiler::recordSample(void* ucontext, u64 counter, EventType event_type, Ev
         num_frames += makeFrame(frames + num_frames, BCI_CPU, java_ctx.cpu | 0x8000);
     }
 
+    VMThread* vm_thread;
+    if (_otel_initialized && getCurrentCompileTask() == NULL && (vm_thread = VMThread::current()) != NULL) {
+        JNIEnv* jni = VM::jni();
+        if (jni != NULL) {
+            jobject current_span = jni->CallStaticObjectMethod(_span_class, _span_current_method);
+            if (current_span != NULL && !jni->ExceptionCheck()) {
+                jobject span_context = jni->CallObjectMethod(current_span, _get_span_context_method);
+                if (span_context != NULL && !jni->ExceptionCheck()) {
+                    jstring trace_id_str = (jstring)jni->CallObjectMethod(span_context, _get_trace_id_method);
+                    jstring span_id_str = (jstring)jni->CallObjectMethod(span_context, _get_span_id_method);
+                    
+                    if (trace_id_str && span_id_str && !jni->ExceptionCheck()) {
+                        const char* trace_id = jni->GetStringUTFChars(trace_id_str, NULL);
+                        const char* span_id = jni->GetStringUTFChars(span_id_str, NULL);
+                        
+                        if (trace_id && span_id) {
+                            printf("TRACE_ID: %s, SPAN_ID: %s\n", trace_id, span_id);
+                        }
+                        
+                        if (trace_id) jni->ReleaseStringUTFChars(trace_id_str, trace_id);
+                        if (span_id) jni->ReleaseStringUTFChars(span_id_str, span_id);
+                    }
+                }
+            }
+            if (jni->ExceptionCheck()) {
+                jni->ExceptionClear();
+            }
+        }
+    }
```

Comment:
This is a lot of JNI calls and string allocations. What is the performance impact? Can we do without jni calls or a single jni call perhaps?